ipn/ipnlocal/local.go

```diff
@@ -2397,9 +2397,20 @@ func peerRoutes(peers []wgcfg.Peer, cgnatThreshold int) (routes []netaddr.IPPref
 	} else {
 		routes = append(routes, cgNATIPs...)
 	}
+
+	sort.Slice(routes, func(i, j int) bool {
+		return ipPrefixLess(routes[i], routes[j])
+	})
```

Comment:
where's the non-determinism in the func above? I see no maps.

Is the input slice random from some caller?

Reply:
> Say node A and B both advertise the same route
> Control picks node A as primary
> A stops advertising route, B is primary now
> This causes ordering in peerIPs to change because the AllowedIPs slice in peers changed
> Note A and B are still connected

Copied from slack for posterity